### PR TITLE
Add streamlabs provider

### PR DESCRIPTION
**Pull request by SertayKabuk**
[Streamlabs](https://dev.streamlabs.com/reference#authorize) provider implemented.

### CODE REVIEW

File: src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationConstants.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            public const string FacebookId = "urn:streamlabs:facebookid";
+            public const string FacebookName = "urn:streamlabs:facebookname";
+            public const string Primary = "urn:streamlabs:primary";
+            public const string Thumbnail = "urn:streamlabs:thumbnail";
+            public const string TwitchDisplayName = "urn:streamlabs:twitchdisplayname";
+            public const string TwitchIconUrl = "urn:streamlabs:twitchiconurl";
+            public const string TwitchId = "urn:streamlabs:twitchid";
+            public const string TwitchName = "urn:streamlabs:twitchname";
+            public const string YoutubeId = "urn:streamlabs:youtubeid";
+            public const string YoutubeTitle = "urn:streamlabs:youtubetitle";
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationConstants.cs]:
```suggestion
            public const string YouTubeId = "urn:streamlabs:youtubeid";
            public const string YouTubeTitle = "urn:streamlabs:youtubetitle";
```

File: src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            ClaimActions.MapCustomJson(Claims.FacebookId, user => GetData(user, "facebook.id"));
+            ClaimActions.MapCustomJson(Claims.FacebookName, user => GetData(user, "facebook.name"));
+            ClaimActions.MapCustomJson(Claims.Primary, user => GetData(user, "streamlabs.primary"));
+            ClaimActions.MapCustomJson(Claims.Thumbnail, user => GetData(user, "streamlabs.thumbnail"));
+            ClaimActions.MapCustomJson(Claims.TwitchDisplayName, user => GetData(user, "twitch.display_name"));
+            ClaimActions.MapCustomJson(Claims.TwitchIconUrl, user => GetData(user, "twitch.icon_url"));
+            ClaimActions.MapCustomJson(Claims.TwitchId, user => GetData(user, "twitch.id"));
+            ClaimActions.MapCustomJson(Claims.TwitchName, user => GetData(user, "twitch.name"));
+            ClaimActions.MapCustomJson(Claims.YoutubeId, user => GetData(user, "youtube.id"));
+            ClaimActions.MapCustomJson(Claims.YoutubeTitle, user => GetData(user, "youtube.title"));
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationOptions.cs]: Looks like all of these can be mapped with `MapJsonSubKey()` and then `GetData()` can be deleted as it becomes redundant.

Here's an example from the Patreon provider:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/0d9f3f21a7561d258b37ceece293df6ed57fc21b/src/AspNet.Security.OAuth.Patreon/PatreonAuthenticationOptions.cs#L32-L37

File: test/AspNet.Security.OAuth.Providers.Tests/Streamlabs/StreamlabsTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        public override string DefaultScheme => StreamlabsAuthenticationDefaults.AuthenticationScheme;
+
+        protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+        {
+            builder.AddStreamlabs(options => ConfigureDefaults(builder, options));
+        }
+
+        [Theory]
```
Comment (martincostello) [inline test/AspNet.Security.OAuth.Providers.Tests/Streamlabs/StreamlabsTests.cs:30]: Please add a test for each claim mapped by default in the options.

File: src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationOptions.cs (hunk 2 above)
Comment (SertayKabuk) [reply src/AspNet.Security.OAuth.Streamlabs/StreamlabsAuthenticationOptions.cs]: This is much better. Thanks for the suggestion. https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/553/commits/3137c89a32a8d232b76e6d2b69058c025ad82033

File: test/AspNet.Security.OAuth.Providers.Tests/Streamlabs/StreamlabsTests.cs (hunk 3 above)
Comment (SertayKabuk) [reply test/AspNet.Security.OAuth.Providers.Tests/Streamlabs/StreamlabsTests.cs:30]: I've added missing tests. https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/553/commits/8500d738ed29df3646cdcb35a97d3a0b9ec6ace3

Sorry for taking your time.

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Provider now available from [NuGet.org](https://www.nuget.org/packages/AspNet.Security.OAuth.Streamlabs/) - thanks again for your contribution.

**Pull request merged**